nemo/collections/asr/models/ctc_models.py

```diff
@@ -237,19 +257,31 @@ def change_vocabulary(self, new_vocabulary: List[str]):
                 zero_infinity=True,
                 reduction=self._cfg.get("ctc_reduction", "mean_batch"),
             )
+
+            if decoding_cfg is None:
+                # Assume same decoding config as before
+                decoding_cfg = self.cfg.decoding
+
+            # Assert the decoding config with all hyper parameters
+            decoding_cls = OmegaConf.structured(CTCCharDecodingConfig)
```

Comment:
Are the structured and to_container calls necessary before using the merge?

Reply:
Yeah. Weirdly, creating omegaconf structured(dataclass) and then trying to merge without below trick, may cause user defined extra overrides to crash the program. Normally, such strictness is good, forcing only valid dataclass values, but in this case, users may pass additional values that the base dataclass does not support, so this trick must be used.